pkg/kubelet/lifecycle/predicate.go

```diff
@@ -72,6 +74,20 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 	pods := attrs.OtherPods
 	nodeInfo := schedulerframework.NewNodeInfo(pods...)
 	nodeInfo.SetNode(node)
+
+	// TODO: Remove this after the SidecarContainers feature gate graduates to GA.
+	if !utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
+		for _, c := range admitPod.Spec.InitContainers {
+			if types.IsSidecarContainer(&c) {
+				return PodAdmitResult{
+					Admit:   false,
+					Reason:  "SidecarForbidden",
+					Message: "An init container cannot have restartPolicy=Always",
```

Comment:
`may not have a non-default restartPolicy` would probably be more accurate.  

Also, I’m not in love with types.IsSidecarContainer - we chose not to call the api sidecar, so the code shouldn’t call it sidecar.  I would prefer:

* `types.IsRestartableInitContainer`
* `InitContainerRestartPolicyForbidden`
* `Init container “%s” has a non-default restartPolicy`

Reply:
> Also, I’m not in love with types.IsSidecarContainer - we chose not to call the api sidecar, so the code shouldn’t call it sidecar. I would prefer:

I think that's a good point.

How about `alwaysRestartableInitContainer`? It is a bit longer, but I think it makes more sense to align with the decision to add the `restartPolicy` field to the init container.

Reply:
will get some more feedback for naming and fix it later.
cc @SergeyKanzhelev

Reply:
Hrm, we should be consistent on naming in the first release in code.  Just because we have a use case for a feature, does not mean we name the code after the feature.  For instance pods support Restart Policy Always, Never, and OnFailure.  OnFailure and Never are completely different lifecycles from Always - the former support batch, the latter support serving workloads.  But we don't have checks in the code `IsBatchPod` or `IsServingPod` - we refer to them in code by their restart behavior.

While the right now we are *enabling* sidecars, the API design was explicitly chosen to allow future use cases such as:

1. init containers running in parallel
2. restart policies that allow containers to shutdown after various lifecycle phases
3. potentially permanent failure init containers ("run exactly one time")

If we add any of those, they wouldn't be "sidecars", and the old name on this check would then be incorrect.  So the core pod loop shouldn't refer to code by use case, but code by the behavior of the API fields.

I will need a very strong argument for using `Sidecar` here to tag this PR, due to precedent in the kubelet and our general motion to improve the readability of the kubelet in general.

Reply:
so this includes the feature gate name `SidecarContainers` ?